UnitTests/View/ViewCommandTests.cs

```diff
@@ -124,6 +124,148 @@ public void MouseClick_Does_Not_Invoke_Accept_Command ()
+        bool A_CancelAccepting = false;
+
+        int B_AcceptedCount = 0;
+        bool B_CancelAccepting = false;
+
+        var w = new Window ()
+        {
+            BorderStyle = LineStyle.None,
+            Width = 10,
+            Height = 10
+        };
+
+        var btnA = new Button ()
+        {
+            Width = 3,
+            IsDefault = true
+        };
+        btnA.Accepting += (s, e) =>
+                          {
+                              A_AcceptedCount++;
+                              e.Cancel = A_CancelAccepting;
+                          };
+
+        var btnB = new Button ()
+        {
+            Width = 3,
+            X = Pos.Right (btnA)
+        };
+
+        btnB.Accepting += (s, e) =>
+                          {
+                              B_AcceptedCount++;
+                              e.Cancel = B_CancelAccepting;
+                          };
+        w.Add (btnA, btnB);
+
+        w.LayoutSubviews ();
+
+        Application.Begin (w);
+        Assert.Same (Application.Top, w);
+
+        // Click button 2
+        var btn2Frame = btnB.FrameToScreen ();
+
+        Application.RaiseMouseEvent (
+                         new MouseEventArgs ()
+                         {
+                             ScreenPosition = btn2Frame.Location,
+                             Flags = MouseFlags.Button1Clicked
+                         });
+
+        // Button A should have been accepted because B didn't cancel and A IsDefault
+        Assert.Equal (1, A_AcceptedCount);
+        Assert.Equal (1, B_AcceptedCount);
+
+        B_CancelAccepting = true;
+        Application.RaiseMouseEvent (
+                                     new MouseEventArgs ()
+                                     {
+                                         ScreenPosition = btn2Frame.Location,
+                                         Flags = MouseFlags.Button1Clicked
+                                     });
+
+        // Button A (IsDefault) should NOT have been accepted because B canceled
+        Assert.Equal (1, A_AcceptedCount);
+        Assert.Equal (2, B_AcceptedCount);
+    }
+
+    // See: https://github.com/gui-cs/Terminal.Gui/issues/3905
+    [Fact]
+    public void Button_CanFocus_False_Raises_Accepted_Correctly ()
+    {
+        int wAcceptedCount = 0;
+        bool wCancelAccepting = false;
+        var w = new Window ()
+        {
+            Title = "Window",
+            BorderStyle = LineStyle.None,
+            Width = 10,
+            Height = 10
+        };
+
+        w.Accepting += (s, e) =>
+                       {
+                           wAcceptedCount++;
+                           e.Cancel = wCancelAccepting;
+                       };
+
+        int btnAcceptedCount = 0;
+        bool btnCancelAccepting = false;
+        var btn = new Button ()
+        {
+            Title = "Button",
+            Width = 3,
+            IsDefault = true,
+        };
+        btn.CanFocus = true;
```

Comment:
`btn.CanFocus` is always `true` here, so I suggest to change to an assertion:

`Assert.True (btn.CanFocus);`

Furthermore, the unit test doesn't really test for what the method name mentions, `Button_CanFocus_False_Raises_Accepted_Correctly` and you really wanted to mean `btn.CanFocus = false;`, which in that case the test will fail.

Reply:
Correct. I forgot to update the test. The test IS named correctly. Now a comment says:

```cs
  // See: https://github.com/gui-cs/Terminal.Gui/issues/3905
  // There is still a bug here, but it's hidden by the fact that Accepting sets e.Cancel to true
  [Fact]
  public void Button_CanFocus_False_Raises_Accepted_Correctly ()
```